src/Renderer/ThreeExtended/FirstPersonControls.js

```diff
@@ -12,15 +10,18 @@ function onDocumentMouseDown(event) {
 
     this._onMouseDownMouseX = event.clientX;
     this._onMouseDownMouseY = event.clientY;
-    this._onMouseDownPhi = this._phi;
-    this._onMouseDownTheta = this._theta;
+    this._onMouseDownRotZ = this.camera.rotation.z;
+    this._onMouseDownRotX = this.camera.rotation.x;
 }
 
 function onDocumentMouseMove(event) {
     if (this._isUserInteracting === true) {
-        const fovCorrection = this.camera.fov / MAX_FOV; // 1 at MAX_FOV
-        this._phi = (this._onMouseDownMouseX - event.clientX) * 0.13 * fovCorrection + this._onMouseDownPhi;
-        this._theta = (event.clientY - this._onMouseDownMouseY) * 0.13 * fovCorrection + this._onMouseDownTheta;
+        // in rigor we have tan(theta) = tan(cameraFOV) * deltaH / H
+        // (where deltaH is the vertical amount we moved, and H the renderer height)
+        // we loosely approximate tan(x) by x
+        const pxToAngleRatio = THREE.Math.degToRad(this.camera.fov) / this.view.mainLoop.gfxEngine.height;
+        this.camera.rotation.z = (event.clientX - this._onMouseDownMouseX) * pxToAngleRatio + this._onMouseDownRotZ;
+        this.camera.rotation.x = (event.clientY - this._onMouseDownMouseY) * pxToAngleRatio + this._onMouseDownRotX;
```

Comment:
I'm not sure about this change. I kinda liked that `camera.matrix` was only modified only in the `mainLoop.update` process.

Reply:
On the other hand, can we require every FrameRequesters not to touch the `camera.matrix` either directly or indirectly? Seems a hard requirement to me. Does this already cause trouble?

Reply:
This code is not run in a FrameRequester context, but in a `mouse-move` event.
Anyway, the real issue is to not modify the camera.matrixWorld between the update and render and this shouldn't happen because events are queued after the current script execution (right?).

So we can probably merge this and gets the benefit of a simplified code.

Reply:
> This code is not run in a FrameRequester context, but in a mouse-move event

Yes sorry, I wasn't clear: can we require client code (whether `GlobeControls` or anything) not to modify the camera outside the update? 

> not modify the camera.matrixWorld between the update and render and this shouldn't happen because events are queued after the current script execution (right?).

Not sure to understand this, setting rotation.{x,y,z} does not modify camera.matrixWorld right?